<a href="https://colab.research.google.com/github/justorfc/estadistica_2025_2/blob/main/Ejercicio_Acero_Grupo_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import os

os.getcwd()

'/content'

In [3]:
os.listdir()

['.config', 'drive', 'sample_data']

In [4]:
os.chdir("/content/drive/MyDrive/Estadística Grupo 2")

In [5]:
os.getcwd()

'/content/drive/MyDrive/Estadística Grupo 2'

In [6]:
os.listdir()

['acero.csv',
 'DATOS_2025_SEM_2.xlsx',
 'Ejercicio Produccion de Acero.docx',
 'Ejercicio Acero Grupo 2.ipynb']

In [9]:
import pandas as pd

df = pd.read_csv("acero.csv")
df

consumo  pr.tbc  pr.cc  pr.ca  pr.galv1  pr.galv2  pr.pint linea hora  \
0    135,31    6840    830      0       579      1401        0     A   1º   
1     84,08     443    903     58       611      1636      717     A   2º   
2    131,62    7270    572     36       982      1963      243     A   3º   
3     90,46    5031    694    122       896      1568        0     A   4º   
4    120,04    9365   1054    157       403      1480        0     A   5º   
..      ...     ...    ...    ...       ...       ...      ...   ...  ...   
112  113,75    9333    455    150       669       715        0     C   3º   
113  105,06    2271      0    391       697      1503        0     C   4º   
114  168,88    9625    112    325       509      1124        0     C   5º   
115  195,61   10979    700    333         0      1319        0     C   6º   
116  213,23   10441    467    248         0      1567        0     C   7º   

    temperatura averias  naverias sistema  ProdTotal     NOx      CO     COV  \
0          Alta      Si         1     OFF      11266    0,49   3,545   0,545   
1          Alta      No         0     OFF       7251  0,0725   2,895   0,425   
2          Baja      No         0     OFF      11066    1,49  5,0075    0,69   
3          Baja      No         0      ON       8311   1,715    2,16    0,36   
4          Baja      No         0     OFF      12459   0,465   4,845  0,6625   
..          ...     ...       ...     ...        ...     ...     ...     ...   
112        Alta      Si         3     OFF      11322   1,025    3,52  0,5225   
113        Alta      No         0     OFF       8939  0,9525   4,005  0,5625   
114       Media      No         0      ON      13845   2,195   5,275   0,745   
115        Baja      No         0     OFF      15906  1,3925  6,8275   0,915   
116       Media      No         0      ON      17247    2,03   7,085  0,9475   

       SO2       CO2   N2O  
0    0,038     101,5  6,35  
1    0,047    63,565  2,23  
2    0,062   98,8175  5,99  
3    0,066   70,1825  3,66  
4    0,086     88,53  6,06  
..     ...       ...   ...  
112   0,06     84,44  4,71  
113  0,104   78,9275  5,39  
114  0,051   128,075  7,44  
115   0,06  146,6625  8,54  
116  0,054   160,385   8,6  

[117 rows x 20 columns]

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype 
---  ------       --------------  ----- 
 0   consumo      117 non-null    object
 1   pr.tbc       117 non-null    int64 
 2   pr.cc        117 non-null    int64 
 3   pr.ca        117 non-null    int64 
 4   pr.galv1     117 non-null    int64 
 5   pr.galv2     117 non-null    int64 
 6   pr.pint      117 non-null    int64 
 7   linea        117 non-null    object
 8   hora         117 non-null    object
 9   temperatura  117 non-null    object
 10  averias      117 non-null    object
 11  naverias     117 non-null    int64 
 12  sistema      117 non-null    object
 13  ProdTotal    117 non-null    int64 
 14  NOx          117 non-null    object
 15  CO           117 non-null    object
 16  COV          117 non-null    object
 17  SO2          117 non-null    object
 18  CO2          117 non-null    object
 19  N2O          117 non-null    

In [11]:
df.describe()

pr.tbc        pr.cc       pr.ca    pr.galv1     pr.galv2  \
count    117.000000   117.000000  117.000000  117.000000   117.000000   
mean    7567.820513   295.529915  124.478632  402.803419  1159.717949   
std     3002.703249   358.618556  161.990772  334.928923   577.791602   
min        0.000000     0.000000    0.000000    0.000000     0.000000   
25%     5950.000000     0.000000    0.000000    0.000000   933.000000   
50%     8808.000000    80.000000    0.000000  432.000000  1360.000000   
75%     9734.000000   582.000000  248.000000  698.000000  1567.000000   
max    10979.000000  1204.000000  677.000000  982.000000  1963.000000   

          pr.pint    naverias     ProdTotal  
count  117.000000  117.000000    117.000000  
mean   188.555556    0.675214  11841.649573  
std    289.445857    1.292078   4075.068276  
min      0.000000    0.000000   2187.000000  
25%      0.000000    0.000000   9023.000000  
50%      0.000000    0.000000  11860.000000  
75%    394.000000    0.000000  14908.000000  
max    898.000000    4.000000  23202.000000

In [12]:
# Leer nuevamente el archivo csv, controlando el punto decimal que aparece como coma

df = pd.read_csv("acero.csv", decimal=",")
df

consumo  pr.tbc  pr.cc  pr.ca  pr.galv1  pr.galv2  pr.pint linea hora  \
0     135.31    6840    830      0       579      1401        0     A   1º   
1      84.08     443    903     58       611      1636      717     A   2º   
2     131.62    7270    572     36       982      1963      243     A   3º   
3      90.46    5031    694    122       896      1568        0     A   4º   
4     120.04    9365   1054    157       403      1480        0     A   5º   
..       ...     ...    ...    ...       ...       ...      ...   ...  ...   
112   113.75    9333    455    150       669       715        0     C   3º   
113   105.06    2271      0    391       697      1503        0     C   4º   
114   168.88    9625    112    325       509      1124        0     C   5º   
115   195.61   10979    700    333         0      1319        0     C   6º   
116   213.23   10441    467    248         0      1567        0     C   7º   

    temperatura averias  naverias sistema  ProdTotal     NOx      CO     COV  \
0          Alta      Si         1     OFF      11266  0.4900  3.5450  0.5450   
1          Alta      No         0     OFF       7251  0.0725  2.8950  0.4250   
2          Baja      No         0     OFF      11066  1.4900  5.0075  0.6900   
3          Baja      No         0      ON       8311  1.7150  2.1600  0.3600   
4          Baja      No         0     OFF      12459  0.4650  4.8450  0.6625   
..          ...     ...       ...     ...        ...     ...     ...     ...   
112        Alta      Si         3     OFF      11322  1.0250  3.5200  0.5225   
113        Alta      No         0     OFF       8939  0.9525  4.0050  0.5625   
114       Media      No         0      ON      13845  2.1950  5.2750  0.7450   
115        Baja      No         0     OFF      15906  1.3925  6.8275  0.9150   
116       Media      No         0      ON      17247  2.0300  7.0850  0.9475   

       SO2       CO2   N2O  
0    0.038  101.5000  6.35  
1    0.047   63.5650  2.23  
2    0.062   98.8175  5.99  
3    0.066   70.1825  3.66  
4    0.086   88.5300  6.06  
..     ...       ...   ...  
112  0.060   84.4400  4.71  
113  0.104   78.9275  5.39  
114  0.051  128.0750  7.44  
115  0.060  146.6625  8.54  
116  0.054  160.3850  8.60  

[117 rows x 20 columns]

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 20 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   consumo      117 non-null    float64
 1   pr.tbc       117 non-null    int64  
 2   pr.cc        117 non-null    int64  
 3   pr.ca        117 non-null    int64  
 4   pr.galv1     117 non-null    int64  
 5   pr.galv2     117 non-null    int64  
 6   pr.pint      117 non-null    int64  
 7   linea        117 non-null    object 
 8   hora         117 non-null    object 
 9   temperatura  117 non-null    object 
 10  averias      117 non-null    object 
 11  naverias     117 non-null    int64  
 12  sistema      117 non-null    object 
 13  ProdTotal    117 non-null    int64  
 14  NOx          117 non-null    float64
 15  CO           117 non-null    float64
 16  COV          117 non-null    float64
 17  SO2          117 non-null    float64
 18  CO2          117 non-null    float64
 19  N2O     

In [14]:
# Estadística descriptiva con la función describe()
df.describe()

consumo        pr.tbc        pr.cc       pr.ca    pr.galv1  \
count  117.000000    117.000000   117.000000  117.000000  117.000000   
mean   139.456496   7567.820513   295.529915  124.478632  402.803419   
std     55.185253   3002.703249   358.618556  161.990772  334.928923   
min     17.500000      0.000000     0.000000    0.000000    0.000000   
25%     99.090000   5950.000000     0.000000    0.000000    0.000000   
50%    140.070000   8808.000000    80.000000    0.000000  432.000000   
75%    182.480000   9734.000000   582.000000  248.000000  698.000000   
max    290.720000  10979.000000  1204.000000  677.000000  982.000000   

          pr.galv2     pr.pint    naverias     ProdTotal         NOx  \
count   117.000000  117.000000  117.000000    117.000000  117.000000   
mean   1159.717949  188.555556    0.675214  11841.649573    1.196132   
std     577.791602  289.445857    1.292078   4075.068276    0.643837   
min       0.000000    0.000000    0.000000   2187.000000    0.012500   
25%     933.000000    0.000000    0.000000   9023.000000    0.775000   
50%    1360.000000    0.000000    0.000000  11860.000000    1.120000   
75%    1567.000000  394.000000    0.000000  14908.000000    1.655000   
max    1963.000000  898.000000    4.000000  23202.000000    2.775000   

               CO         COV         SO2         CO2         N2O  
count  117.000000  117.000000  117.000000  117.000000  117.000000  
mean     4.377350    0.620983    0.066504  104.628141    6.114786  
std      2.095038    0.255447    0.022064   41.389709    2.125250  
min      0.250000    0.025000    0.001000   14.285000    0.870000  
25%      2.927500    0.452500    0.054000   73.717500    4.710000  
50%      4.107500    0.625000    0.066000  105.307500    6.170000  
75%      5.585000    0.785000    0.080000  134.877500    7.580000  
max     10.170000    1.255000    0.127000  218.307500   11.140000

La salida de la función `describe()` proporciona un resumen estadístico de las columnas numéricas en el DataFrame.

**Interpretación de las métricas:**

*   **count**: Número de observaciones no nulas en cada columna. En este caso, todas las columnas numéricas tienen 117 observaciones, lo que indica que no hay valores faltantes en estas columnas.
*   **mean**: El valor promedio de cada columna. Por ejemplo, el consumo promedio es de aproximadamente 139.46 y la producción total promedio es de alrededor de 11841.65.
*   **std**: La desviación estándar, que mide la dispersión o variabilidad de los datos. Una desviación estándar alta indica que los valores están más dispersos del promedio, mientras que una baja indica que están más cerca del promedio. Por ejemplo, el consumo tiene una desviación estándar de 55.19, mientras que `pr.tbc` tiene una desviación estándar mucho mayor de 3002.70.
*   **min**: El valor mínimo en cada columna.
*   **25% (Primer Cuartil)**: El valor por debajo del cual se encuentra el 25% de los datos.
*   **50% (Mediana)**: El valor medio de los datos cuando están ordenados. También conocido como el segundo cuartil.
*   **75% (Tercer Cuartil)**: El valor por debajo del cual se encuentra el 75% de los datos.
*   **max**: El valor máximo en cada columna.

Este resumen estadístico es útil para obtener una primera idea de la distribución de los datos, identificar posibles valores atípicos (observando los valores mínimos y máximos en relación con los cuartiles y la media) y comprender la variabilidad de cada variable numérica.

Aquí tienes algunas sugerencias de análisis estadísticos que se pueden realizar con el archivo `acero.csv`, basándote en la descripción de las variables:

**Análisis Descriptivo:**

*   **Resumen estadístico:** Ya has realizado esto con `df.describe()`, lo cual es un excelente primer paso para entender la distribución de las variables numéricas.
*   **Análisis de frecuencias:** Para las variables categóricas (`linea`, `hora`, `temperatura`, `averias`, `sistema`), se pueden obtener tablas de frecuencia para ver la distribución de las observaciones en cada categoría.
*   **Visualizaciones:** Crear histogramas para las variables numéricas para visualizar su distribución, y gráficos de barras para las variables categóricas. Box plots pueden ser útiles para identificar valores atípicos.

**Análisis de Relación entre Variables:**

*   **Correlación:** Calcular la matriz de correlación para las variables numéricas para identificar relaciones lineales entre ellas. Esto puede ayudar a ver, por ejemplo, cómo se relaciona el consumo energético con los diferentes tipos de producción.
*   **Análisis de la relación entre el consumo energético y otras variables:**
    *   **Consumo vs. Producción:** Investigar si existe una relación entre el consumo energético (`consumo`) y los diferentes tipos de producción (`pr.tbc`, `pr.cc`, `pr.ca`, `pr.galv1`, `pr.galv2`, `pr.pint`, `ProdTotal`). Se podrían usar diagramas de dispersión y análisis de regresión simple o múltiple.
    *   **Consumo vs. Variables Categóricas:** Analizar si el consumo energético varía significativamente según la línea de producción, la hora, la temperatura, la presencia de averías o la activación del sistema de sobrecalentamiento. Se podrían usar pruebas t (si hay dos categorías) o ANOVA (si hay más de dos categorías).
*   **Análisis de la relación entre averías y otras variables:**
    *   **Averías vs. Producción:** ¿Las averías (`averias`, `naverias`) afectan los niveles de producción?
    *   **Averías vs. Temperatura/Sistema:** ¿La temperatura o la activación del sistema de sobrecalentamiento están relacionadas con la ocurrencia de averías?

**Modelado Predictivo:**

*   **Predicción del consumo energético:** Se podría construir un modelo de regresión para predecir el consumo energético (`consumo`) en función de las otras variables disponibles (producción, línea, hora, temperatura, averías, sistema). Esto podría ayudar a identificar los principales impulsores del consumo energético.
*   **Predicción de averías:** Si el objetivo es predecir la probabilidad de que ocurra una avería, se podría utilizar un modelo de clasificación.

**Análisis Específicos basados en la Descripción:**

*   **Análisis por línea de producción:** Comparar el consumo energético y los patrones de producción entre las diferentes líneas (A, B, C).
*   **Análisis por hora:** Investigar cómo varían el consumo energético y la producción a lo largo de las horas del turno.
*   **Análisis de la última hora:** Dado que la última hora tiene menos mediciones, podría ser interesante analizar si hay alguna diferencia notable en el comportamiento de las variables en comparación con las otras horas.

Estas son solo algunas ideas para empezar. La elección de los análisis específicos dependerá de los objetivos particulares de la investigación.